File: APIs/atletas - JPA/src/main/java/com/academia/atletas/model/Atleta.java
package com.academia.atletas.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;

@Entity
public class Atleta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @NotNull
    private String nome;
    private LocalDate dataAssinatura;
    private LocalDate dataDeVencimento;
    @ManyToOne
    @JoinColumn(name = "professor_id")
    @JsonIgnoreProperties("alunos")
    private Professor professor;

    public Atleta() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public LocalDate getDataAssinatura() {
        return dataAssinatura;
    }

    public void setDataAssinatura(LocalDate dataAssinatura) {
        this.dataAssinatura = dataAssinatura;
    }

    public LocalDate getDataDeVencimento() {
        return dataDeVencimento;
    }

    public void setDataDeVencimento(LocalDate dataDeVencimento) {
        this.dataDeVencimento = dataDeVencimento;
    }

    public Professor getProfessor() {
        return professor;
    }

    public void setProfessor(Professor professor) {
        this.professor = professor;
    }
}

File: frontend/atleta-frontend/src/app/models/Atleta.ts
import { Professor} from './Professor';
export class Atleta{
    id?: number;
    nome!: string;
    dataAssinatura!: Date;
    dataDeVencimento!: Date;
    professor?: {
        id: number,
        nome: string
        alunos: Atleta[]
    } = {} as Professor
    

    constructor(nome: string, dataAssinatura: Date, dataDeVencimento: Date,id?: number, professor?: Professor){
        this.id = id;
        this.nome = nome;
        this.dataAssinatura = dataAssinatura;
        this.dataDeVencimento = dataDeVencimento;
        this.professor = professor;
    }
}